pkg/azure/aks/machine_store.go

```diff
@@ -125,16 +141,44 @@ func (m *MachineStore) getVmInfoFromVmss(ctx context.Context, rgName, vmssName,
 				continue
 			}
 
-			vmInfo[vmName] = &VirtualMachineInfo{
-				Name:            vmName,
-				Region:          to.String(v.Location),
-				OwningVMSS:      vmssName,
-				OwningCluster:   cluster,
-				MachineTypeSku:  to.String(v.SKU.Name),
-				Priority:        priority,
-				OperatingSystem: osInfo,
+			if v.SKU == nil || v.SKU.Name == nil {
+				m.logger.LogAttrs(ctx, slog.LevelDebug, "no VM Sku found", slog.String("machineName", vmName))
+				continue
+			}
+			vmSku := to.String(v.SKU.Name)
+
+			vmRegion := to.String(v.Location)
+			if len(vmRegion) == 0 {
+				m.logger.LogAttrs(ctx, slog.LevelDebug, "no VM region found", slog.String("machineName", vmName))
+				continue
+			}
+
+			if v.Properties == nil || v.Properties.VMID == nil {
+				m.logger.LogAttrs(ctx, slog.LevelDebug, "no VM ID found", slog.String("machineName", vmName))
+				continue
 			}
-			m.logger.LogAttrs(ctx, slog.LevelDebug, "found machine information", slog.String("machineName", vmName))
+			vmId := to.String(v.Properties.VMID)
+
+			if vmSizeInfo, ok := m.MachineSizeMap[vmRegion][vmSku]; ok {
+				vmInfo[vmId] = &VirtualMachineInfo{
+					Name:            vmName,
+					Id:              vmId,
+					Region:          vmRegion,
+					OwningVMSS:      vmssName,
+					OwningCluster:   cluster,
+					MachineTypeSku:  vmSku,
+					Priority:        priority,
+					OperatingSystem: osInfo,
+
+					NumOfCores:     to.Int32(vmSizeInfo.NumberOfCores),
+					MemoryInMB:     to.Int32(vmSizeInfo.MemoryInMB),
+					OsDiskSizeInMB: to.Int32(vmSizeInfo.OSDiskSizeInMB),
+				}
+				m.logger.LogAttrs(ctx, slog.LevelDebug, "found machine information", slog.String("machineName", vmName))
+				continue
+			}
```

Comment:
Suggestion, keep the happy path outside of the if statement if possible. It's a bit confusing as I'm reviewing to see all of the other if statements handle error, then this one handles the "ok" path.

Reply:
updated! WDYT?